Chat bot 2.0

mgrant46 opened the pull request:
Ishruna approved the video. Wondering how to insert the intro video on the lesson plan.

mgrant46 committed:
Update to Chatbot 2.0

There is skeleton of the format. Also includes a basic "rule based activity"
More content will be added

mgrant46 committed:
Updates

Fixed typos

mgrant46 committed:
Updated

add Callysto Banner on the bottom.

byrcyb reviewed (commented):
Things are looking good. Here are my comments / suggestions:
- an accompanying lesson plan document similar to the other ones on the Callysto website (https://www.callysto.ca/lesson-plans/) or providing more details in the notebook would be useful in understanding how to use the activities describing the notebook, and the set-up required (e.g. laptop, paper, pen). For example for activity 1, students should start by forming teams of 2 to 3 where each person has the role of Turing tester, human or robot. Perhaps the activity should be called Tester, Human or Robot?
- Replace `Intro to Chatbot including AI information: TBD` with `#### Play the video below for an introduction to Chatbots and Artificial Intelligence.` Below this add a Python code cell to embed the youtube video once there is a url. The format can look like this but replace the video id 
```
from IPython.display import YouTubeVideo
YouTubeVideo('-rb_w6G1Cq0')
```
- Suggestion: `- What does the following chatbots respond well? ` replace with `Which prompts did the chatbot respond well to?`
- for Activity 1, I got a bit confused reading the instructions. a clearer mock-up describing would be nice perhaps using the role names. For example:
```
Round 1:
Decide who will be the Tester. The remaining two players decide who is Human or Robot in secret.

Tester: Says "Hi" as a prompt.
Robot: Chooses a chatbot, writes "Hi" in chat dialog and repeats aloud the response from the chatbot.
Human: Responds to the "Hi" prompt. 

Tester then guesses who is Robot or Human. If they guess both correctly they get a point. Record your score.

Round 2:
Tester remains the same.
Remaining players decide whether or not to keep the same role.

Tester: Gives a new prompt.
Human: Responds to prompt.
Robot: Uses the same chatbot, writes prompt in chat dialog and repeats aloud the response from the chatbot.

Tester then guesses who is Robot or Human. If they guess both correctly they get a point. Record your score.

Repeat for 5 rounds. 
Then switch Tester role with remaining players, where each person gets the Tester role for 5 rounds. The person with the most points wins after all players get a turn as Tester wins.
Make note of any prompts that the chat bot performed well on in terms of making the Tester think it was a human. Also, make note of which chat bot performed well in terms of making the Tester think it was human.
```
- In the code block for activity 2, add code comments using # to indicate where in the code students can modify the code to make it more engaging. For example:
```
menu = ["menu item1", "menu item2", "menu item3",
        "menu item4", "menu item5", "menu item6"]        #replace menu itemX with your menu item
```
- For the conclusion, do we want to ask them to record their responses / reflections somewhere and share with the teacher?

mgrant46 committed:
Updates

Added more description to the lesson

mgrant46 committed:
Title Change

This will likely a beginning lesson for chatbots. Therefore changed the name from 2.0 to 1.0

mgrant46 committed:
Activity 2

Adding more description in the instructions.

mgrant46 committed:
Activity 1 video added

Added a descriptive video explaining a turing test.
https://www.youtube.com/watch?v=4VROUIAF2Do

mgrant46 committed:
Activity 2

Added comments to the code to match the required tasks

byrcyb committed:
Update youtube video link

byrcyb committed:
Add more explicit instructions

byrcyb committed:
change notebook name

byrcyb committed:
rename
